app/experimenter/static/rapid/components/experiments/ExperimentDetails.tsx

```diff
@@ -58,6 +64,15 @@ const ExperimentDetails: React.FC = () => {
 
   const handleClickRequestApproval = async () => dispatch(requestReview());
 
+  React.useEffect(() => {
+    const visualizationRoot = document.getElementById("visualization-root");
+    if (!visualizationRoot || !window.initSvelte) {
+      return;
+    }
+
+    window.initSvelte(visualizationRoot);
+  });
```

Comment:
Oh there's a thing with useEffect.  By default it runs on every render:

`
The default behavior for effects is to fire the effect after every completed render. That way an effect is always recreated if one of its dependencies changes.
`

But you can pass it a second arg which is the value it checks for changes:

https://reactjs.org/docs/hooks-reference.html#conditionally-firing-an-effect

You have it setup now to just return if it's already initialized, but the pattern to make it only run once at startup is to pass an empty array:

`
If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument. 
`

so I think if you do that you don't need the conditional check and it'll only run once?

Reply:
I'm not sure if this matters right now since we're replacing `rapid` soon, but we shouldn't use `document.getElementById` in a React component, using a [ref](https://reactjs.org/docs/refs-and-the-dom.html) is the way to reference elements👍

Reply:
Oh neat!

Reply:
So, I've updated this to use a ref. However, using the `[]` as a second param for `useEffect()` causes it to only be called once before rendering I think, because the ref is not set at that point.